tests/test_http.py

```diff
@@ -235,3 +236,30 @@ def test_django_json_encoder(self):
             response["HX-Trigger"]
             == '{"showMessage": {"uuid": "12345678-1234-5678-1234-567812345678"}}'
         )
+
+    def test_custom_json_encoder(self):
+        class BeanCounter:
+            def __init__(self, magic_bean_count: int):
+                self.magic_bean_count = magic_bean_count * 2
+
+        class MyEncoder(DjangoJSONEncoder):
+            def default(self, o):
+                if isinstance(o, BeanCounter):
+                    return {"magic_beans": o.magic_bean_count}
+
+                return super().default(o)
+
+        response = HttpResponse()
+        uuid_value = UUID("{12345678-1234-5678-1234-567812345678}")
+
+        trigger_client_event(
+            response,
+            "showMessage",
+            {"uuid": uuid_value, "beans": BeanCounter(1)},
+            encoder=MyEncoder,
+        )
+
+        assert response["HX-Trigger"] == (
+            '{"showMessage": {"uuid": "12345678-1234-5678-1234-567812345678", '
+            '"magic_beans": 2}}'
+        )
```

Comment:
This test can be shortened a bit, we don't need a UUID or the bean count to prove the encoder is used